### PR TITLE
Add signature validation action

### PR DESCRIPTION
**Pull request by Qix-**
I offer this to help the maintainers a bit; feel free to shoot it down, won't bother me a bit.

This is an action that validates PRs and performs a strict check against signatures added to the README.

- Green checkmark means the PR strictly passes validation; it follows the correct format, is only one line, and the changes are only within the README. It also makes sure that the pull requester's username matches the one added to the README.
- Red X (failure) doesn't necessarily mean anything is wrong, just that maintainers should take a closer look.

For example, this PR will "fail" simply because it's not a single line change. It just means maintainers should look a bit closer at the PR.

I see the PRs piling up so I figured this might cut down the overhead slightly.

### CODE REVIEW

**Comment by drop-ice** [comment]
This is great! Is the current failure just because it is on a branch? It doesn't look like it's throwing the right error

**Comment by Qix-** [comment]
@drop-ice good catch. It's now "failing" for the right reason :)

**Comment by Qix-** [comment]
Now checks for alphabetical order (totally forgot). Caveat is that it only checks that it's in alphabetical order given the previous and the next line, so the readme will need to be sorted for this to be useful.

**Comment by Qix-** [comment]
Alright, cleaned up and should be good to go. I'll monitor any failures after merging and make sure it's working correctly :)

**Comment by drop-ice** [comment]
This is fabulous, thank you so much!

File: .github/workflows/validate.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	console.error('Added line to beginning of README.');
+	process.exit(1);
+}
+
+if (addedIndex === chunks[0].changes.length - 1) {
+	console.error('Added line to end of README (not able to automatically check)');
+	process.exit(1);
+}
+
+const linePattern = /^\- ([^,]+), @([^ ,]+)(?: \([^)]+\))?$/;
```
Comment (drop-ice) [inline .github/workflows/validate.js]: is this checking for alphabetization by first name? we're doing it by last name. noticed it failing consistently

**Comment by drop-ice** [comment]
@Qix- I reverted to avoid confusing people til we can fix the alphabetization check

File: .github/workflows/validate.js (hunk 1 above)
Comment (Qix-) [reply .github/workflows/validate.js]: Ah, yes.

**Pull request merged**